test/vrf.js

```diff
@@ -1,29 +1,41 @@
 const VRF = artifacts.require("VRF")
-const testdata = require("./testdata.json")
+const truffleAssert = require("truffle-assertions")
 
 contract("VRF", accounts => {
+  const data = require("./data.json")
+
   describe("Auxiliary functions: ", () => {
     let vrf
     before(async () => {
       vrf = await VRF.deployed()
     })
-    for (let [index, proof] of testdata.proofs.valid.entries()) {
+    for (let [index, test] of data.proofs.valid.entries()) {
       it(`should decode a VRF proof from bytes (${index + 1})`, async () => {
-        const decodedProof = await vrf.decodeProof.call(web3.utils.hexToBytes(proof.pi))
-        assert(decodedProof[0].eq(web3.utils.toBN(proof.gamma.x)))
-        assert(decodedProof[1].eq(web3.utils.toBN(proof.gamma.y)))
-        assert(decodedProof[2].eq(web3.utils.toBN(proof.c)))
-        assert(decodedProof[3].eq(web3.utils.toBN(proof.s)))
+        const decodedProof = await vrf.decodeProof.call(web3.utils.hexToBytes(test.pi))
+        assert(decodedProof[0].eq(web3.utils.toBN(test.gamma.x)))
+        assert(decodedProof[1].eq(web3.utils.toBN(test.gamma.y)))
+        assert(decodedProof[2].eq(web3.utils.toBN(test.c)))
+        assert(decodedProof[3].eq(web3.utils.toBN(test.s)))
+      })
+    }
+    for (let [index, test] of data.proofs.invalid.entries()) {
+      it(`should not decode a malformed VRF proof from bytes (${index + 1}) - ${test.description}`, async () => {
+        await truffleAssert.reverts(vrf.decodeProof.call(web3.utils.hexToBytes(test.pi)), test.revert)
       })
     }
-    for (let [index, point] of testdata.points.valid.entries()) {
+    for (let [index, test] of data.points.valid.entries()) {
       it(`should decode a compressed EC Point (${index + 1})`, async () => {
-        const coord = await vrf.decodePoint.call(web3.utils.hexToBytes(point.compressed))
-        assert(coord[0].eq(web3.utils.toBN(point.uncompressed.x)))
-        assert(coord[1].eq(web3.utils.toBN(point.uncompressed.y)))
+        const coord = await vrf.decodePoint.call(web3.utils.hexToBytes(test.compressed))
+        assert(coord[0].eq(web3.utils.toBN(test.uncompressed.x)))
+        assert(coord[1].eq(web3.utils.toBN(test.uncompressed.y)))
       })
     }
-    for (let [index, test] of testdata.fastVerify.valid.entries()) {
+    for (let [index, test] of data.points.invalid.entries()) {
+      it(`should not decode a malformed compressed EC Point (${index + 1}) - ${test.description}`, async () => {
```

Comment:
Negative tests can be better named by the source of the failure. E.g. `should fail to decode a compressed EC point if malformed`